### ISSUE TITLE
How do I use toEmails with the array data type?

### ISSUE THREAD

**Issue by powderblock** [simalexan/api-lambda-send-email-ses#6]
I am trying to test my deployed serverless app in Postman.

Here is my body
![body](https://user-images.githubusercontent.com/1458607/84295231-531a0380-ab18-11ea-896e-942063ee8266.png)

And here is the response:
![response](https://user-images.githubusercontent.com/1458607/84295249-57deb780-ab18-11ea-9039-390af82a2656.png)

I can't figure out what I'm doing wrong. Started over from the beginning and redeployed but it's not working. Ideas?

Thanks in advance!

**Comment by powderblock** [comment]
This seems to be resolved by passing in an array for toEmails param. Further testing required on my end. My god, is the paint tool fun to write with. LOL. 

![different input different output](https://user-images.githubusercontent.com/1458607/84296299-d720bb00-ab19-11ea-81fd-1740f190db3d.png)

**Comment by qrs17** [comment]
Are you able to test this email function successfully from within Lambda? I am trying to test using the exact same JSON as you show above, but the script keeps acting like the body is empty.

**Comment by kerem66** [comment]
Any solution about that topic? Getting the same error forwarding me to see the logs.

**Comment by kerem66** [comment]
Okay found how to send the email for verified emails. I checked the logs you need a verified account to send and receive the email. Both receiver and sender should be verified. You can go here https://console.aws.amazon.com/ses/home, just make sure that your deployment region and selected region on the top right corner are same and on the left panel click "email adresses" and on the screen select "verify email adress" and check for the verification email. After that you can send email to that mail adress. If you want to send to unverified receivers check that link https://docs.aws.amazon.com/ses/latest/DeveloperGuide/request-production-access.html

**Comment by tahaerden** [comment]
Hello, I'm having the same problem. Any help would be appreciated.

In SES I have 2 verified identities:
1. First one is a domain (info@mywebsite.com). It is verified, but it is waiting for review for production access.
2. Second one is my personal email address to be used as a receiver. I verified this receiver address based on @kerem66 's suggestion.

I am able to send emails from SES console. But when I call the API I get "Error: Execution update, caused a SES error, please look at your logs."

**Comment by kerem66** [comment]
Can u see any specific error in your logs ? If so, can u share it? If you can't see any , can u share your email sender code.

**Comment by tahaerden** [comment]
@kerem66 thanks for the quick reply.

For the logs, I checked CloudWatch -> Logs -> Log groups ->serverlessrepo-api-lambda-send-email-LambdaEmailer. (please let me know if I should check anywhere else)

Here I see an error as follows:

ERROR	AccessDenied: User `arn:aws:sts::595648071683:assumed-role/serverlessrepo-api-lambda-send-e-LambdaEmailerRole-1VPXPP01VOM9T/serverlessrepo-api-lambda-send-email-LambdaEmailer-BjodeMOxukQw' is not authorized to perform `ses:SendEmail' on resource 

I am using Amplify with Angular/Ionic. I am guessing my amplify web client doesn't have permission to use this Lambda function? After deploying this serverless component, I didn't manually add any permissions. Please tell me how should I proceed, I am quite new to AWS environment.

Here is my code to send the email:

            async sendEmail() {
                const url = 'https://<SOMETHING>.execute-api.eu-central-1.amazonaws.com/Prod/send';
                const body = {
                toEmails: ['myEmailAddress@something.com'],
                subject: 'Test subject',
                message: 'test message'
                };
                const res = await this.http.post(url, body).toPromise();
                console.log(res);
            }

**Comment by kerem66** [comment]
Yes, you need to give permission to specific lambda function which is triggered from this endpoint "'https://<SOMETHING>.execute-api.eu-central-1.amazonaws.com/Prod/send';" .  You can use this link to follow the steps -> https://aws.amazon.com/tr/premiumsupport/knowledge-center/lambda-send-email-ses/ . To explain it; you need to create IAM policy to attach sending email permissions-> https://console.aws.amazon.com/iam , go to roles tab and create a new role, choose lambda as use case, click next, on the permissions tab search with "ses" and add "AmazonSESFullAccess" also add "AWSLambdaBasicExecutionRole" permission to that role, then next next until you enter a role name, give a reasonable name for that role and the role will be created. After that we need to attach this role to lambda, Go to your lambda https://console.aws.amazon.com/lambda/home this link will redirect you to lambdas page i guess. select your lambda and go to configuration tab and click to edit button on the execution role section 
![image](https://user-images.githubusercontent.com/26746679/122590352-1d342700-d06a-11eb-9d57-fcaf35f4e575.png)
After that on the existing role part click the button and you will see the roles, select the role we created before and save, that should work i guess. 
![image](https://user-images.githubusercontent.com/26746679/122590515-5a001e00-d06a-11eb-8475-5f3409e6157e.png)

**Comment by tahaerden** [comment]
@kerem66 thanks a lot for the detailed explanation. It works now!

@simalexan the default IAM role had AWSLambdaBasicExecutionRole and LambdaEmailerRolePolicy0 (Pinpoint Email, SES, SES v2). Now thanks to @kerem66 's suggestion, I used a custom role having AmazonSESFullAccess and AWSLambdaBasicExecutionRole and it worked. So maybe you should update the default IAM roles?
